### PR TITLE
feat(timezone-picker): Timezone Picker

### PR DESCRIPTION
**Pull request by chasestarr**
#### Description

Screen Shot 2019-04-15 at 12 13 49 PM

some interesting reading related to timezones/date-time programming:
https://zachholman.com/talk/utc-is-enough-for-everyone-right
https://medium.com/@toastui/handling-time-zone-in-javascript-547e67aa842d

#### Scope

- [x] Minor: New Feature

### CODE REVIEW

**Comment by tajo** [comment]
Looking great! Should we provide an option to translate labels through some optional function?

**Comment by chasestarr** [comment]
> Looking great! Should we provide an option to translate labels through some optional function?

Yes - I will edit this pr today

File: src/datepicker/types.js [hunk 1]
```diff
@@ -270,6 +270,11 @@ export type TimezonePickerPropsT = {
    * exact specificity, provide a date. Otherwise it will default to the relevant timezone at render.
    */
   date?: Date,
+  /**
+   * Customize the option's label. Useful for translations and optionally mapping from
+   * 'America/Los_Angeles' to 'Pacific Time'.
+   */
+  mapLabels?: OptionT => string,
```
Comment (tajo) [inline src/datepicker/types.js]: Any reason why to pass the whole `OptionT` and not just the label? Also, the select's label is type `React.Node`?

Comment (chasestarr) [reply src/datepicker/types.js]: good point about `React.Node` will update. 

in this case, the `id` is the IANA title (`America/Los_Angeles`), and the label contains some additional information like the offset, and short code (`PST`/`PDT`). I thought that extra context may be useful

Comment (chasestarr) [reply src/datepicker/types.js]:
```suggestion
  mapLabels?: OptionT => React.Node,
```

**Comment by chasestarr** [comment]
Just realized that I need to handle the `date` prop update case. Will update

**Comment by chasestarr** [comment]
can't reproduce the puppeteer fail locally, gonna look into this today

**Pull request merged**